src/components/surveys/survey-design/question-edit/rhs-subcontrols/Year.test.tsx

```diff
@@ -32,87 +52,201 @@ const step: YearQuestion = {
+      await user.keyboard('2039')
+    })
+    expect(maxYear.value).toHaveValue(2039)
+  })
+
+  test('error is shown when allowPast is false and maximumYear is less than current year', async () => {
+    const currentYear = new Date().getFullYear()
+    const errorYear = currentYear - 1
+
+    const {user, maxYear} = setUp(createStepProps({allowPast: false, maximumYear: currentYear}))
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+
+    await waitFor(async () => {
+      await user.dblClick(maxYear.value!)
+      await user.keyboard(errorYear.toString())
+    })
+    expect(maxYear.value).toHaveValue(errorYear)
+    const alert = screen.queryByRole('alert')
+    expect(alert).toBeInTheDocument()
+    expect(alert).toHaveTextContent(ErrorMessages['NO_PAST_YEARS'])
+  })
+
+  test('error is shown when allowFuture is false and minimumYear is greater than current year', async () => {
+    const currentYear = new Date().getFullYear()
+    const errorYear = currentYear + 1
+
+    const {user, minYear} = setUp(createStepProps({allowFuture: false, minimumYear: currentYear}))
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+
+    await waitFor(async () => {
+      await user.dblClick(minYear.value!)
+      await user.keyboard(errorYear.toString())
+    })
+    expect(minYear.value).toHaveValue(errorYear)
+    const alert = screen.queryByRole('alert')
+    expect(alert).toBeInTheDocument()
+    expect(alert).toHaveTextContent(ErrorMessages['NO_FUTURE_YEARS'])
+  })
+
+  test('error is shown when maximumYear is less than minimumYear', async () => {
+    const year = 2023
+    const errorYear = year - 1
+    const {user, minYear, maxYear} = setUp(createStepProps({minimumYear: year, maximumYear: year}))
+
+    expect(minYear.value).toHaveValue(year)
+    expect(maxYear.value).toHaveValue(year)
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+
+    await waitFor(async () => {
+      await user.dblClick(maxYear.value!)
+      await user.keyboard(errorYear.toString())
+    })
+    expect(maxYear.value).toHaveValue(errorYear)
+    const alert = screen.queryByRole('alert')
+    expect(alert).toBeInTheDocument()
+    expect(alert).toHaveTextContent(ErrorMessages['RANGE'])
+  })
+
+  test('error is shown when minimumYear is greater than maximumYear', async () => {
+    const year = 2023
+    const errorYear = year + 1
+    const {user, minYear, maxYear} = setUp(createStepProps({minimumYear: year, maximumYear: year}))
+
+    expect(minYear.value).toHaveValue(year)
+    expect(maxYear.value).toHaveValue(year)
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+
+    await waitFor(async () => {
+      await user.dblClick(minYear.value!)
+      await user.keyboard(errorYear.toString())
+    })
+    expect(minYear.value).toHaveValue(errorYear)
+    const alert = screen.queryByRole('alert')
+    expect(alert).toBeInTheDocument()
+    expect(alert).toHaveTextContent(ErrorMessages['RANGE'])
+  })
+})
```

Comment:
Nice set of tests. You could consider adding a test for the stale data bugfix if you implement a fix for the `Year` component.